apps/block_scout_web/lib/block_scout_web/controllers/chain/market_history_chart_controller.ex

```diff
@@ -0,0 +1,51 @@
+defmodule BlockScoutWeb.Chain.MarketHistoryChartController do
+  use BlockScoutWeb, :controller
+
+  alias Explorer.{Chain, Market}
+  alias Explorer.ExchangeRates.Token
+
+  def show(conn, _params) do
+    with true <- ajax?(conn) do
+      exchange_rate = Market.get_exchange_rate(Explorer.coin()) || Token.null()
+
+      market_history_data =
+        30
+        |> Market.fetch_recent_history()
+        |> case do
+          [today | the_rest] -> [%{today | closing_price: exchange_rate.usd_value} | the_rest]
+          data -> data
+        end
+        |> encode_market_history_data
```

Comment:
```suggestion
        |> encode_market_history_data()
```

Just to keep the pattern some lines above.